kytos/core/interface.py

```diff
@@ -173,65 +181,332 @@ def status_reason(self):
+        ranges_a: list[list[int]],
+        ranges_b: list[list[int]]
+    ) -> list[list[int]]:
+        """Returns the intersection between two list
+        of ranges"""
+        result = []
+        a_i, b_i = 0, 0
+        while a_i < len(ranges_a) and b_i < len(ranges_b):
+            fst_a, snd_a = ranges_a[a_i]
+            fst_b, snd_b = ranges_b[b_i]
+            # Moving forward with non-intersection
+            if snd_a < fst_b:
+                a_i += 1
+            elif snd_b < fst_a:
+                b_i += 1
+            else:
+                # Intersection
+                intersection_start = max(fst_a, fst_b)
+                intersection_end = min(snd_a, snd_b)
+                result.append([intersection_start, intersection_end])
+                if snd_a < snd_b:
+                    a_i += 1
+                else:
+                    b_i += 1
+        return result
+
+    @staticmethod
+    def range_intersection_first_tag(
+        ranges_a: list[list[int]],
+        ranges_b: list[list[int]]
+    ) -> Optional[int]:
+        """Returns an intersection tag between two list
+        of ranges"""
+        a_i, b_i = 0, 0
+        while a_i < len(ranges_a) and b_i < len(ranges_b):
+            fst_a, snd_a = ranges_a[a_i]
+            fst_b, snd_b = ranges_b[b_i]
+
+            # Moving forward with non-intersection
+            if snd_a < fst_b:
+                a_i += 1
+            elif snd_b < fst_a:
+                b_i += 1
+            else:
+                # Intersection
+                intersection_tag = max(fst_a, fst_b)
+                return intersection_tag
 
-        Args:
-            iterable ([int]): range of VLANs.
+        return None
+
+    @staticmethod
+    def range_difference(
+        ranges_a: list[list[int]],
+        ranges_b: list[list[int]]
+    ) -> list[list[int]]:
+        """The operation is (ranges_a - ranges_b).
+        This method simulates difference of sets. E.g.:
+        [[10, 15]] - [[4, 11], [14, 45]] = [[12, 13]]
         """
+        result = []
+        a_i, b_i = 0, 0
+        update = True
+        while a_i < len(ranges_a) and b_i < len(ranges_b):
+            if update:
+                start_a, end_a = ranges_a[a_i]
+            else:
+                update = True
+            start_b, end_b = ranges_b[b_i]
+
+            # Moving forward with non-intersection
+            if end_a < start_b:
+                result.append([start_a, end_a])
+                a_i += 1
+            elif end_b < start_a:
+                b_i += 1
+            else:
+                # Intersection
+                if start_a < start_b:
+                    result.append([start_a, start_b - 1])
+
+                if end_a > end_b:
+                    start_a = end_b + 1
+                    update = False
+                    b_i += 1
+                else:
+                    a_i += 1
+
+        # Append the rest of ranges_a
+        while a_i < len(ranges_a):
+            if update:
```

Comment:
I see that the comment is misleading. Apart from adding the rest of `ranges_a` also adds the last intersection. For example:
```
a = [[10, 15], [30, 31]]
b = [[4, 11]]
```
Without the update statement this method results in `[[10, 15], [30, 31]]` instead of `[[12, 15], [30, 31]]`